examples/plant_location2.py

```diff
@@ -0,0 +1,219 @@
+r"""Example with Special Ordered Sets (SOS) type 1 and 2.
+Plant location: one industry plans to install two plants in two different
+regions, one to the west and another to the east. It must decide also the
+production capacity of each plant and allocate clients to plants
+in order to minimize shipping costs. We'll use SOS of type 1 to ensure that
+only one of the plants in each region has a non-zero production capacity.
+The cost :math:`f(z)` of building a plant with capacity :math:`z` grows according
+to the non-linear function :math:`f(z)=1520 \log z`. Type 2 SOS will be used to
+model the cost of installing each one of the plants.
+"""
+
+from __future__ import annotations
+
+import sys
+
+# Workaround for issues with python not being installed as a framework on mac
+# by using a different backend.
+if sys.platform == "darwin":  # OS X
+    import matplotlib as mpl
+    mpl.use('Agg')
+    del mpl
+
+import math
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import mip
+
+if TYPE_CHECKING:
```

Comment:
@mwerezak I never have seen this construct before. From my best knowledge, someone should not use abc things for typing

Reply:
I see now it's not compatible before Python 3.9.

Reply:
I see now that compatibility before Python 3.9 depends on the tooling.
https://peps.python.org/pep-0585/#backwards-compatibility

Reply:
@sebheger Is there a way to run the CI locally?